### PR TITLE
BAEL-1980 - Sharing H2 database

### PR DESCRIPTION
**Pull request by micropatel**
patelmayurm@gmail.com

### CODE REVIEW

File: spring-boot-h2/spring-boot-h2-database/pom.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<groupId>com.baeldung.h2db</groupId>
+	<artifactId>spring-boot-h2-database</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<packaging>jar</packaging>
+
+	<name>spring-boot-app1</name>
```
Comment (KevinGilmore) [inline spring-boot-h2/spring-boot-h2-database/pom.xml]: guideline states to omit the name element

File: spring-boot-h2/spring-boot-h2-database/pom.xml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<groupId>com.baeldung.h2db</groupId>
```
Comment (KevinGilmore) [inline spring-boot-h2/spring-boot-h2-database/pom.xml]: indent with spaces, not tabs

File: spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import org.h2.tools.Server;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+@SpringBootApplication
+public class SpringBootApp {
+
```
Comment (KevinGilmore) [inline spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java]: use the standard formatter (if using Eclipse) to ensure the indentation meets the requirements as stated in the guidelines (for example, use spaces, not tabs)

File: spring-boot-h2/spring-boot-h2-remote-app/pom.xml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<groupId>com.baeldung.h2db</groupId>
+	<artifactId>spring-boot-h2-remote-app</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<packaging>jar</packaging>
+
+	<name>spring-boot-app2</name>
```
Comment (KevinGilmore) [inline spring-boot-h2/spring-boot-h2-remote-app/pom.xml]: remove the name element

File: spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java (hunk 3 above)
Comment (micropatel) [reply spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java]: I am using Intelli J with Google Code Style

Comment (KevinGilmore) [reply spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java]: It's definitely similar to the style we're using. There is a standard formatter available at https://github.com/eugenp/tutorials/tree/master/intelliJ that you can try.
If you can't get that to work, then the main difference I see is in the xml files, where the standard indentation is four spaces per level, and this one seems to be using two spaces.

File: spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                                rs.getString("id"),
+                                rs.getString("first_name"),
+                                rs.getString("last_name")));
+                        return null;
+                    }
+                });
+    }
+
+    @Bean(initMethod = "start", destroyMethod = "stop")
+    public Server inMemoryH2DatabaseaServer() throws SQLException {
```
Comment (KevinGilmore) [inline spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java]: spelling in method name "...Databasea..."

File: spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        SpringApplication.run(SpringBootApp.class, args);
+    }
+
+    @PostConstruct
+    private void initDb() {
+        System.out.println(String.format(
+                "****** Creating table: %s, and Inserting test data ******", "Employees"));
+
+        String sqlStatements[] = {
+                "drop table employees if exists",
```
Comment (KevinGilmore) [inline spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java]: please correct the indentation in this section

File: spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        };
+
+        Arrays.asList(sqlStatements).stream().forEach(sql -> {
+            System.out.println(sql);
+            jdbcTemplate.execute(sql);
+        });
+
+        System.out.println(String.format("****** Fetching from table: %s ******", "Employees"));
+        jdbcTemplate.query("select id,first_name,last_name from employees",
+                new RowMapper<Object>() {
```
Comment (KevinGilmore) [inline spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java]: please correct indentation here as well

File: spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java (hunk 6 above)
Comment (micropatel) [reply spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java]: fixed.

File: spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java (hunk 7 above)
Comment (micropatel) [reply spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java]: fixed

File: spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java (hunk 5 above)
Comment (micropatel) [reply spring-boot-h2/spring-boot-h2-database/src/main/java/com/baeldung/h2db/demo/SpringBootApp.java]: fixed

File: spring-boot-h2/spring-boot-h2-remote-app/src/main/java/com/baeldung/h2db/demo/ClientSpringBootApp.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+    public static void main(String[] args) {
+        SpringApplication.run(ClientSpringBootApp.class, args);
+    }
+
+    @PostConstruct
+    private void initDb() {
+        System.out.println("****** Inserting more sample data in the table: Employees ******");
+        String sqlStatements[] = {
+                "insert into employees(first_name, last_name) values('Donald','Trump')",
```
[the remark this answers is not in the document]
Comment (micropatel) [reply spring-boot-h2/spring-boot-h2-remote-app/src/main/java/com/baeldung/h2db/demo/ClientSpringBootApp.java]: fixed

**Pull request merged**